### ISSUE TITLE
Fix false positives for leading single-line comment in selector-descendant-combinator-in-non-space

### ISSUE THREAD

**Issue by xereda** [stylelint/stylelint#3429]
Stylelint is presenting a false-positive type inconsistency for the `selector-descendant-combinator-in-non-space` rule. This happens when we add a single line comment. Please, review this ilustration:

![_falso-positivo](https://user-images.githubusercontent.com/14337850/42035178-b46e3422-7ab8-11e8-839f-3ab3ec8339e4.gif)

> Which rule, if any, is the bug related to?

`selector-descendant-combinator-no-non-space`

> What CSS is needed to reproduce the bug?

```vue
<template>
  <span class="test">
    { test }
  </span>
</template>

<script>
export default {
  name: 'test',
  data() {
    return {
      test: 'hello world!',
    };
  },
};
</script>

<style lang="scss" scoped>
// one line comments
.test {
  margin: 10px;
}
</style>

```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "processors": ["stylelint-processor-html"],
  "plugins": [
    "stylelint-order",
    "stylelint-scss"
  ],
  "rules": {
    "scss/dollar-variable-pattern": "^foo",
    "scss/selector-no-redundant-nesting-selector": true,
    "order/order": [
			"custom-properties",
			"declarations"
		],
		"order/properties-alphabetical-order": true,
    "at-rule-empty-line-before": [
        "always",
        {
            "except": [
                "blockless-after-same-name-blockless",
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "at-rule-name-case": "lower",
    "at-rule-name-space-after": "always-single-line",
    "at-rule-semicolon-newline-after": "always",
    "block-closing-brace-empty-line-before": "never",
    "block-closing-brace-newline-after": "always",
    "block-closing-brace-newline-before": "always-multi-line",
    "block-closing-brace-space-before": "always-single-line",
    "block-opening-brace-newline-after": "always-multi-line",
    "block-opening-brace-space-after": "always-single-line",
    "block-opening-brace-space-before": "always",
    "color-hex-case": "lower",
    "color-hex-length": "short",
    "comment-empty-line-before": [
        "always",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "stylelint-commands"
            ]
        }
    ],
    "comment-whitespace-inside": "always",
    "custom-property-empty-line-before": [
        "always",
        {
            "except": [
                "after-custom-property",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "declaration-bang-space-after": "never",
    "declaration-bang-space-before": "always",
    "declaration-block-semicolon-newline-after": "always-multi-line",
    "declaration-block-semicolon-space-after": "always-single-line",
    "declaration-block-semicolon-space-before": "never",
    "declaration-block-single-line-max-declarations": 1,
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-newline-after": "always-multi-line",
    "declaration-colon-space-after": "always-single-line",
    "declaration-colon-space-before": "never",
    "declaration-empty-line-before": [
        "always",
        {
            "except": [
                "after-declaration",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "function-comma-newline-after": "always-multi-line",
    "function-comma-space-after": "always-single-line",
    "function-comma-space-before": "never",
    "function-max-empty-lines": 0,
    "function-name-case": "lower",
    "function-parentheses-newline-inside": "always-multi-line",
    "function-parentheses-space-inside": "never-single-line",
    "function-whitespace-after": "always",
    "indentation": 2,
    "length-zero-no-unit": true,
    "max-empty-lines": 1,
    "media-feature-colon-space-after": "always",
    "media-feature-colon-space-before": "never",
    "media-feature-name-case": "lower",
    "media-feature-parentheses-space-inside": "never",
    "media-feature-range-operator-space-after": "always",
    "media-feature-range-operator-space-before": "always",
    "media-query-list-comma-newline-after": "always-multi-line",
    "media-query-list-comma-space-after": "always-single-line",
    "media-query-list-comma-space-before": "never",
    "no-eol-whitespace": true,
    "no-missing-end-of-source-newline": true,
    "number-leading-zero": "never",
    "number-no-trailing-zeros": true,
    "property-case": "lower",
    "property-no-unknown": true,
    "rule-empty-line-before": [
        "always-multi-line",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "selector-attribute-brackets-space-inside": "never",
    "selector-attribute-operator-space-after": "never",
    "selector-attribute-operator-space-before": "never",
    "selector-combinator-space-after": "always",
    "selector-combinator-space-before": "always",
    "selector-descendant-combinator-no-non-space": true,
    "selector-list-comma-newline-after": "always",
    "selector-list-comma-space-before": "never",
    "selector-max-empty-lines": 0,
    "selector-pseudo-class-case": "lower",
    "selector-pseudo-class-parentheses-space-inside": "never",
    "selector-pseudo-element-case": "lower",
    "selector-pseudo-element-colon-notation": "double",
    "selector-type-case": "lower",
    "unit-case": "lower",
    "value-list-comma-newline-after": "always-multi-line",
    "value-list-comma-space-after": "always-single-line",
    "value-list-comma-space-before": "never",
    "value-list-max-empty-lines": 0
  }
}
```

> Which version of stylelint are you using?

```json
    "stylelint": "^9.3.0",
    "stylelint-order": "^0.8.1",
    "stylelint-processor-html": "^1.0.0",
    "stylelint-scss": "^3.1.3",
```

> How are you running stylelint: CLI, PostCSS plugin, Node API?

With vscode.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

SCSS in Vue.js files (*.vue). 

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warning is flagged:

![captura de tela de 2018-06-28 09-54-31](https://user-images.githubusercontent.com/14337850/42035424-481d892a-7ab9-11e8-9e4a-f074fa878631.png)

**Comment by jeddy3** [comment]
@xereda Thanks for the report and for using the template.

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by BreizhReloaded** [comment]
@xereda Hi! May I ask how you manage to use Style-Lint within a Vue component file? I'm using VSCode and have been trying to find something for a few hours now, so I'm quite interested. Thanks!

**Comment by hudochenkov** [comment]
@BreizhReloaded https://github.com/shinnn/vscode-stylelint

**Comment by xereda** [comment]
@BreizhReloaded Hi! I did it with the sum of the following extensions: vscode-stylelint + vetur. If you want, I can send the vscode configuration file.

**Comment by BreizhReloaded** [comment]
@hudochenkov Already went there and installed it, but it was not that easy!
@xereda I'm using this combo as well, but I couldn't make it for now. I'm sending you an email, thank you very much!

**Comment by xereda** [comment]
Esqueci uma coisa. Eu criei o arquivo de configuração do stylelint na raiz do projeto (`.stylelint`).

```json

{
  "plugins": [
    "stylelint-order",
    "stylelint-scss",
    "stylelint-declaration-use-variable",
  ],
  "rules": {
    "sh-waqar/declaration-use-variable": "/color/",
    "scss/selector-no-redundant-nesting-selector": true,
    "order/order": [
			"custom-properties",
			"declarations"
		],
		"order/properties-alphabetical-order": true,
    "at-rule-empty-line-before": [
        "always",
        {
            "except": [
                "blockless-after-same-name-blockless",
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "at-rule-name-case": "lower",
    "at-rule-name-space-after": "always-single-line",
    "at-rule-semicolon-newline-after": "always",
    "block-closing-brace-empty-line-before": "never",
    "block-closing-brace-newline-after": "always",
    "block-closing-brace-newline-before": "always-multi-line",
    "block-closing-brace-space-before": "always-single-line",
    "block-opening-brace-newline-after": "always-multi-line",
    "block-opening-brace-space-after": "always-single-line",
    "block-opening-brace-space-before": "always",
    "color-named": "never",
    "color-hex-case": "lower",
    "color-hex-length": "short",
    "comment-empty-line-before": [
        "always",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "stylelint-commands"
            ]
        }
    ],
    "comment-whitespace-inside": "always",
    "custom-property-empty-line-before": [
        "always",
        {
            "except": [
                "after-custom-property",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "declaration-bang-space-after": "never",
    "declaration-bang-space-before": "always",
    "declaration-block-semicolon-newline-after": "always-multi-line",
    "declaration-block-semicolon-space-after": "always-single-line",
    "declaration-block-semicolon-space-before": "never",
    "declaration-block-single-line-max-declarations": 1,
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-newline-after": "always-multi-line",
    "declaration-colon-space-after": "always-single-line",
    "declaration-colon-space-before": "never",
    "declaration-empty-line-before": [
        "always",
        {
            "except": [
                "after-declaration",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "function-comma-newline-after": "always-multi-line",
    "function-comma-space-after": "always-single-line",
    "function-comma-space-before": "never",
    "function-max-empty-lines": 0,
    "function-name-case": "lower",
    "function-parentheses-newline-inside": "always-multi-line",
    "function-parentheses-space-inside": "never-single-line",
    "function-whitespace-after": "always",
    "indentation": 2,
    "length-zero-no-unit": true,
    "max-empty-lines": 1,
    "media-feature-colon-space-after": "always",
    "media-feature-colon-space-before": "never",
    "media-feature-name-case": "lower",
    "media-feature-parentheses-space-inside": "never",
    "media-feature-range-operator-space-after": "always",
    "media-feature-range-operator-space-before": "always",
    "media-query-list-comma-newline-after": "always-multi-line",
    "media-query-list-comma-space-after": "always-single-line",
    "media-query-list-comma-space-before": "never",
    "no-eol-whitespace": true,
    "no-missing-end-of-source-newline": true,
    "number-leading-zero": "never",
    "number-no-trailing-zeros": true,
    "property-case": "lower",
    "property-no-unknown": true,
    "rule-empty-line-before": [
        "always-multi-line",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "selector-attribute-brackets-space-inside": "never",
    "selector-attribute-operator-space-after": "never",
    "selector-attribute-operator-space-before": "never",
    "selector-combinator-space-after": "always",
    "selector-combinator-space-before": "always",
    "selector-descendant-combinator-no-non-space": true,
    "selector-list-comma-newline-after": "always",
    "selector-list-comma-space-before": "never",
    "selector-max-empty-lines": 0,
    "selector-pseudo-class-case": "lower",
    "selector-pseudo-class-parentheses-space-inside": "never",
    "selector-pseudo-element-case": "lower",
    "selector-pseudo-element-colon-notation": "double",
    "selector-type-case": "lower",
    "unit-case": "lower",
    "value-list-comma-newline-after": "always-multi-line",
    "value-list-comma-space-after": "always-single-line",
    "value-list-comma-space-before": "never",
    "value-list-max-empty-lines": 0
  }
}


```

**Comment by evilebottnawi** [comment]
@xereda please use English language

**Comment by xereda** [comment]
@evilebottnawi ?

**Comment by ota-meshi** [comment]
I do not think you need to use `stylelint-processor-html` to lint vue files.
It worked correctly by deleting `stylelint-processor-html` from the configuration.
